protos/file.proto

```diff
@@ -19,7 +19,191 @@
 syntax = "proto3";
 
 package lance.file;
+
+// Lance v2 File Format
+// 
+// +------------------------------+
+// | Page 0, Buffer 0             |
+// | ...                          |
+// | Page 0, Buffer N             |
+// | (optional padding)           |
+// | Page 1, Buffer 0             |
+// | ...                          |
+// | Page N2, Buffer N3           |
+// | Column 0 Metadata            |
+// | Column 0 Metadata Buffer 0   |
+// | ...                          |
+// | Column 0 Metadata Buffer N4  |
+// | (optional padding)           |
+// | Column 1 Metadata            |
+// | ...                          |
+// | Column N5 Metadata Buffer N6 |
+// | (optional padding)           |
+// | Column 0 Metadata Position   |
+// | Column 0 Metadata Size       |
+// | ...                          |
+// | Column N5 Metadata Position  |
+// | Column N5 Metadata Size      |
```

Comment:
This is the "column metadata offsets array"?

Reply:
Yes